api/v1alpha4/machinedeployment_webhook_test.go

```diff
@@ -86,6 +97,58 @@ func TestMachineDeploymentValidation(t *testing.T) {
 			labels:    map[string]string{"-123-foo": "bar"},
 			expectErr: true,
 		},
+		{
+			name:      "should return error for invalid maxSurge",
+			selectors: map[string]string{"-123-foo": "bar"},
+			labels:    map[string]string{"-123-foo": "bar"},
+			strategy: MachineDeploymentStrategy{
+				Type: RollingUpdateMachineDeploymentStrategyType,
+				RollingUpdate: &MachineRollingUpdateDeployment{
+					MaxUnavailable: &goodMaxUnavailableInt,
+					MaxSurge:       &badMaxSurge,
+				},
+			},
+			expectErr: true,
+		},
+		{
+			name:      "should return error for invalid maxUnavailable",
+			selectors: map[string]string{"-123-foo": "bar"},
+			labels:    map[string]string{"-123-foo": "bar"},
+			strategy: MachineDeploymentStrategy{
+				Type: RollingUpdateMachineDeploymentStrategyType,
+				RollingUpdate: &MachineRollingUpdateDeployment{
+					MaxUnavailable: &badMaxUnavailable,
+					MaxSurge:       &goodMaxSurgeInt,
+				},
+			},
+			expectErr: true,
+		},
+		{
+			name:      "should not return error for valid int maxSurge and maxUnavailable",
+			selectors: map[string]string{"-123-foo": "bar"},
+			labels:    map[string]string{"-123-foo": "bar"},
+			strategy: MachineDeploymentStrategy{
+				Type: RollingUpdateMachineDeploymentStrategyType,
+				RollingUpdate: &MachineRollingUpdateDeployment{
+					MaxUnavailable: &goodMaxUnavailableInt,
+					MaxSurge:       &goodMaxSurgeInt,
+				},
+			},
+			expectErr: true,
```

Comment:
Maybe I"m missing something but the combination of "should not return error" and `expectErr: true` is either not correct or not intuitive. (Same below)

Reply:
My bad I was carrying over bad labels and therefore failing always. Fixed, thanks!